src/Transform.jl

```diff
@@ -40,17 +33,14 @@ function logit{NumType <: Number}(x::NumType)
 end
 
 
-function logit{NumType <: Number}(x::Array{NumType})
-    1.0 ./ (1.0 + exp(-x))
-end
-
-
 """
 Convert an (n - 1)-vector of real numbers to an n-vector on the simplex, where
 the last entry implicitly has the untransformed value 1.
 """
 function constrain_to_simplex{NumType <: Number}(x::Vector{NumType})
     if any(x .== Inf)
+        # Is there a bug here? If more than 1 entry in x is Inf, that just means
+        # that the last entry in z is 0, not the both entries in z are the same.
```

Comment:
I think `Inf`s are always bad news, even just one, a `NaN` waiting to happen downstream. Even if `Inf` is the "right" answer, you've got to worry about numerical stability. Two `Inf`s here would be especially problematic. Maybe we should enforce box constraints (for both simplex variables and others) by excluding out-of-bounds regions of the domain from the trust region. We could continue to use the `simplex_box` transform. Exponentiation leads to some really poorly conditioned parts of the domain. What do you think?